src/shared_memory_manager.h

```diff
@@ -90,11 +132,19 @@ class SharedMemoryManager {
   /// \param memory_type Returns the type of the memory
   /// \param device_id Returns the device id associated with the
   /// memory block
-  /// \return a TRITONSERVER_Error indicating success or failure.
+  /// \param shm_info_ref Returns a shared pointer reference(read-only) to the
+  /// shared memory block's information.
+  /// This pointer will automatically increase the usage count, preventing
+  /// unregistration while the reference is held. The reference must be cleared
+  /// or set to nullptr when no longer needed, to decrease the count and allow
+  /// unregistration.
+  /// \return a TRITONSERVER_Error indicating success or
+  /// failure.
   TRITONSERVER_Error* GetMemoryInfo(
       const std::string& name, size_t offset, size_t byte_size,
       void** shm_mapped_addr, TRITONSERVER_MemoryType* memory_type,
-      int64_t* device_id);
+      int64_t* device_id,
+      std::shared_ptr<const SharedMemoryInfo>* shm_info_ref);
```

Comment:
I think that `shm_info` already contains information that covers other args
```suggestion
      const std::string& name, 
      std::shared_ptr<const SharedMemoryInfo>* shm_info_ref);
```

Reply:
Yes, Guan. I think if we only return `shm_info_ref` from this method, we might need to move the [validations and shm_mapped_address casting code](https://github.com/triton-inference-server/server/blob/main/src/shared_memory_manager.cc#L472-L502) out of this method. This could result in multiple places where we call `GetMemoryInfo()` and need to rewrite these validations. Or I think we can write another method (as part of this class) that takes `shm_info_ref` as input and performs these validations.
Could you please let me know if you have any thoughts on this.
Thank you.

Reply:
I see. Yes then it's better to just keep what you have for now, as we are intended to return a partial buffer of the shared memory based on the request.